README.md
mx-findshares
===================

GUI wrapper for findshares, program for installing searching shares on the network


debian/changelog
mx-findshares (19.7) mx; urgency=medium

  * fix window buttons for Buster

 -- Adrian <adrian@mxlinux.org>  Sun, 07 Jul 2019 13:41:07 -0400

mx-findshares (19.4) mx; urgency=medium

  * updated translations

 -- Adrian <adrian@mxlinux.org>  Fri, 05 Apr 2019 11:32:43 -0400

mx-findshares (19.1) mx; urgency=medium

  * updated translations

 -- Adrian <adrian@mxlinux.org>  Wed, 23 Jan 2019 09:50:55 -0500

mx-findshares (18.11) mx; urgency=medium

  * updated translations

 -- Adrian <adrian@mxlinux.org>  Fri, 30 Nov 2018 12:21:27 -0500

mx-findshares (18.10) mx; urgency=medium

  * updated translations

 -- Adrian <adrian@mxlinux.org>  Sun, 14 Oct 2018 18:45:47 -0400

mx-findshares (18.8) mx; urgency=medium

  * fix icon other themes

 -- Adrian <adrian@mxlinux.org>  Sat, 11 Aug 2018 16:20:41 -0400

mx-findshares (18.6) mx; urgency=medium

  * new translations

 -- Adrian <adrian@mxlinux.org>  Mon, 18 Jun 2018 14:23:23 -0400

mx-findshares (18.2mx16) mx; urgency=medium

  * new translations

 -- Adrian <adrian@mxlinux.org>  Sat, 10 Feb 2018 01:46:26 -0500

mx-findshares (17.6mx16) mx; urgency=medium

  * updated translations

 -- Adrian <adrian@mxlinux.org>  Fri, 02 Jun 2017 19:08:51 -0400

mx-findshares (17.5mx16) mx; urgency=medium

  * new icons

 -- Adrian <adrian@mxlinux.org>  Wed, 10 May 2017 11:58:08 -0400

mx-findshares (17.3mx16) mx; urgency=medium

  * new translations

 -- Adrian <adrian@mxlinux.org>  Thu, 30 Mar 2017 10:06:34 -0400

mx-findshares (16.11mx16+1) mx; urgency=medium

  * FR translation

 -- Adrian <adrian@mxlinux.org>  Wed, 30 Nov 2016 12:01:10 -0500

mx-findshares (16.11mx16) mx; urgency=medium

  * new translations

 -- Adrian <adrian@mxlinux.org>  Wed, 16 Nov 2016 13:00:25 -0500

mx-findshares (16.10mx16) mx; urgency=medium

  * changed URLs

 -- Adrian <adrian@mxlinux.org>  Mon, 10 Oct 2016 13:18:36 -0400

mx-findshares (16.6mx15) mx; urgency=medium

  * minor change

 -- Adrian <adrian@mxlinux.org>  Mon, 06 Jun 2016 18:17:37 -0400

mx-findshares (16.5mx15) mx; urgency=medium

  * new desktop file

 -- dolphinoracle <dolphinoracle@gmail.com>  Mon, 23 May 2016 13:11:51 -0400

mx-findshares (15.12mx15) mx; urgency=medium

  * hide GUI when displaying help/license

 -- Adrian <adrian@mxlinux.org>  Tue, 22 Dec 2015 21:54:46 -0500

mx-findshares (15.11mx15) mx; urgency=medium

  * new translations

 -- Adrian <adrian@mxlinux.org>  Mon, 30 Nov 2015 11:51:37 -0500

mx-findshares (15.10.2mx15) mx; urgency=medium

  * more translations

 -- Adrian <adrian@mxlinux.org>  Sat, 24 Oct 2015 19:15:54 -0400

mx-findshares (15.10.1mx15) mx; urgency=medium

  * new translations

 -- Adrian <adrian@mxlinux.org>  Fri, 23 Oct 2015 22:27:39 -0400

mx-findshares (15.10mx15) mx; urgency=medium

  * changed icon

 -- Adrian <adrian@mxlinux.org>  Sun, 18 Oct 2015 15:32:40 -0400

mx-findshares (15.7mx14) mx; urgency=medium

  * update to qt5

 -- Adrian <adrian@mxlinux.org>  Tue, 30 Jun 2015 12:53:03 -0400

mx-findshares (15.6mx14) mx; urgency=medium

  * changed documentation URL

 -- Adrian <adrian@mxlinux.org>  Wed, 03 Jun 2015 11:20:48 -0400

mx-findshares (15.4mx14) mx; urgency=medium

  * added translation

 -- Adrian <adrian@mxlinux.org>  Tue, 21 Apr 2015 14:03:30 -0400

mx-findshares (15.3mx14) mx; urgency=medium

  * change manual URL

 -- Adrian <adrian@mxlinux.org>  Fri, 20 Mar 2015 16:38:02 -0400

mx-findshares (15.2) mx; urgency=medium

  * changed manual URL

 -- Adrian <adrian@mxlinux.org>  Sun, 15 Feb 2015 10:49:14 -0500

mx-findshares (14.3.1) mx; urgency=medium

  * Minor fixes.

 -- Adrian <adrian@mxlinux.org>  Sat, 06 Dec 2014 00:50:25 -0500

mx-findshares (14.2+2) mx; urgency=medium

  * Upgraded findshares to v1.06

 -- Adrian <adrian@mxlinux.org>  Tue, 07 Oct 2014 12:41:54 -0400

mx-findshares (14.2+1) mx; urgency=low

  * Initial release. 

 -- Adrian <adrian@mxlinux.org>  Sun, 28 Sep 2014 12:37:25 -0400


findshares-src/packetdata.h
/* Packed data structure templates for network packets*/

#define B char
#define W uint16_t
#define L uint32_t
#define D uint64_t
#define swap_16(x) (((x) & 0x00ff) << 8 | ((x) & 0xff00) >> 8)
#define swap_32(x) (((x) & 0x00ff) << 24 | ((x) & 0xff00) << 8 | ((x) & 0xff0000) >> 8 | ((x) & 0xff000000) >> 24)

#include "netrshareenum.h"
#include "rxpackettemplates.h"
#include "NFSpackets.h"

static struct frame0
{
W tid;
W mode;
W qdcount;
W ancount;
W nscount;
W arcount;
B queryname[34];
W nbstat;
W in;
}__attribute__((__packed__)) statreq={ .qdcount=swap_16(0x0001),
  .queryname=" CKAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA",
  .in=swap_16(0x0001) };

static struct frame1
{
W sessionflags;
W length;
B callee[34];
B caller[34];
}__attribute__((__packed__)) sessionreq={ .sessionflags=swap_16(0x8100),
  .length=swap_16(0x0044) };

static struct frame2
{
W sessionflags;
W length;
L ffsmb_string;
B proto_command;
B error_class;
B reserved;
W error_code;
B flags;
W flags2;
W pid_hi;
D signature;
W reserved2;
W tid;
W pid;
W uid;
W muxid;
B wordcount;
W bytecount;
B dialect;			// Always 2. Must preceed each protocol string listed.
B string[11];		// First and in our case only protocol string.
}__attribute__((__packed__)) protocol={ .length=swap_16(47),
  .ffsmb_string=0x424D53FF, .proto_command=0x72, .flags=0x08, .flags2=0x0001,
  .tid=0xFFFF, .muxid=2, .bytecount=12, .dialect=2,
  .string[0]="NT LM 0.12" };

static struct frame3
{
W sessionflags;
W length;
L ffsmb_string;
B setup_command;
B error_class;
B reserved;
W error_code;
B flags;
W flags2;
W pid_hi;
D signature;
W reserved2;
W tid;
W pid;
W uid;
W muxid;
B wordcount;
B AndXCommand;
B reserved3;
W AndXOffset;
W maxbuffer;
W maxmpxcount;
W vcnumber;
L sessionkey;
W ansipasswordlen;
W unicodepasswordlen;
L reserved4;
L capabilities;
W bytecount;
B ansipassword;
B strings[64];
}__attribute__((__packed__)) setup={ .ffsmb_string=0x424D53FF,
  .setup_command=0x73, .flags=0x08, .flags2=0x0001,
  .muxid=3, .wordcount=13,
  .AndXCommand=0xFF, .maxbuffer=1024, .maxmpxcount=2, 
  .ansipasswordlen=1, .capabilities=0x00000203, .strings[0]="Guest" };

static struct frame4
{
W sessionflags;
W length;
L ffsmb_string;
B conn_command;
B error_class;
B reserved;
W error_code;
B flags;
W flags2;
W pid_hi;
D signature;
W reserved2;
W tid;
W pid;
W uid;
W muxid;
B wordcount;
B AndXCommand;
B reserved3;
W AndXOffset;
W flags3;
W passwordlen;
W bytecount;
B password;
B strings[32];
}__attribute__((__packed__)) treeconn={ .ffsmb_string=0x424D53FF,
  .conn_command=0x75, .flags=0x08, .flags2=0x0001,
  .tid=0xFFFF, .muxid=4, .wordcount=4,
  .AndXCommand=0xFF, .passwordlen=1, .bytecount=32, .strings[0]="\\\\" };


static struct frame5
{
W sessionflags;
W length;
L ffsmb_string;
B transaction_command;
B error_class;
B reserved;
W error_code;
B flags;
W flags2;
W pid_hi;
D signature;
W reserved2;
W tid;
W pid;
W uid;
W muxid;
B wordcount;
W totparamcnt;
W totdatacnt;
W maxparamcnt;
W maxdatacnt;
B maxsetupcnt;
B reserved3;
W flags3;
L timeout;
W reserved4;
W parametercount;
W parameteroffset;
W datacount;
W dataoffset;
B setupcount;
B reserved5;
W bytecount;
B transaction_name[13];
W functioncode;
B paramdesc[6];
B returndesc[7];
W detail_level;
W rxbufferlen;
}__attribute__((__packed__)) enumshares={ .length=swap_16(95),
  .ffsmb_string=0x424D53FF, .transaction_command=0x25, .flags=0x08,
  .flags2=0x0001, .muxid=6, .wordcount=14,
  .totparamcnt=19, .maxparamcnt=64, .maxdatacnt=4000, .flags3=0x0000,
  .parametercount=19, .parameteroffset=76, .dataoffset=95, .bytecount=32,
  .transaction_name="\\PIPE\\LANMAN", .paramdesc="WrLeh",
  .returndesc="B13BWz", .detail_level=1, .rxbufferlen=4000 };


static struct frame6
{
W sessionflags;
W length;
L ffsmb_string;
B transaction_command;
B error_class;
B reserved;
W error_code;
B flags;
W flags2;
W pid_hi;
D signature;
W reserved2;
W tid;
W pid;
W uid;
W muxid;
B wordcount;
W totparamcnt;
W totdatacnt;
W maxparamcnt;
W maxdatacnt;
B maxsetupcnt;
B reserved3;
W flags3;
L timeout;
W reserved4;
W parametercount;
W parameteroffset;
W datacount;
W dataoffset;
B setupcount;
B reserved5;
W bytecount;
B transaction_name[13];
W functioncode;
B paramdesc[8];
B returndesc[8];
W detail_level;
W rxbufferlen;
L servertype;
B workgroup[16];
}__attribute__((__packed__)) enumservers2={ .length=swap_16(118),
  .ffsmb_string=0x424D53FF, .transaction_command=0x25, .flags=0x08, .flags2=0x0001,
  .muxid=8, .wordcount=14,
  .totparamcnt=40, .maxparamcnt=8, .maxdatacnt=4000, .flags3=0x0001,
  .parametercount=40, .parameteroffset=76, .dataoffset=116, .bytecount=53,
  .transaction_name="\\PIPE\\LANMAN", .functioncode=104, .paramdesc="WrLehDz",
  .returndesc="B16BBDz", .detail_level=1, .rxbufferlen=4000, .servertype=0xffffffff };

// This call is not supported by Samba

static struct frame6A
{
W sessionflags;
W length;
L ffsmb_string;
B transaction_command;
B error_class;
B reserved;
W error_code;
B flags;
W flags2;
W pid_hi;
D signature;
W reserved2;
W tid;
W pid;
W uid;
W muxid;
B wordcount;
W totparamcnt;
W totdatacnt;
W maxparamcnt;
W maxdatacnt;
B maxsetupcnt;
B reserved3;
W flags3;
L timeout;
W reserved4;
W parametercount;
W parameteroffset;
W datacount;
W dataoffset;
B setupcount;
B reserved5;
W bytecount;
B transaction_name[13];
//B Padding;			// Used to word align Unicode strings
//B transaction_name[26];
//W Padding2;			// Used to word align Unicode strings
W functioncode;
B paramdesc[5];
B returndesc[8];
W detail_level;
W rxbufferlen;
// Non Unicode version
}__attribute__((__packed__)) netservergetinfo={ .length=swap_16(95),
  .ffsmb_string=0x424D53FF, .transaction_command=0x25, .flags=0x08,
  .flags2=0x0001, .muxid=9, .wordcount=14,
  .totparamcnt=19, .maxparamcnt=16, .maxdatacnt=4000, .flags3=0x0000,
  .parametercount=19, .parameteroffset=76, .dataoffset=95, .bytecount=32,
  .transaction_name="\\PIPE\\LANMAN", .functioncode=13, .paramdesc="WrLh",
  .returndesc="B16BBDz", .detail_level=1, .rxbufferlen=4000 };

/* 
// Unicode version
}__attribute__((__packed__)) netservergetinfo={ .length=swap_16(111),
  .ffsmb_string=0x424D53FF, .transaction_command=0x25, .flags=0x08,
  .flags2=0x8001, .muxid=9, .wordcount=14,
  .totparamcnt=19, .maxparamcnt=6, .maxdatacnt=4000, .flags3=0x0000,
  .parametercount=19, .parameteroffset=92, .dataoffset=111, .bytecount=48,
  .transaction_name="\\\0P\0I\0P\0E\0\\\0L\0A\0N\0M\0A\0N\0\0", .functioncode=13, .paramdesc="WrLh",
  .returndesc="B16BBDz", .detail_level=1, .rxbufferlen=4000 };
*/

#undef B
#undef W
#undef L
#undef D
#undef swap_16
#undef swap_32
